srtcore/core.cpp

```diff
@@ -3682,11 +3685,21 @@ void srt::CUDT::startConnect(const sockaddr_any& serv_addr, int32_t forced_isn)
                 cst = processRendezvous(&response, serv_addr, RST_OK, (reqpkt));
                 if (cst == CONN_CONTINUE)
                     continue;
-                break;
+
+                // rejection or erroneous code.
+                reqpkt.setLength(m_iMaxSRTPayloadSize);
+                reqpkt.setControl(UMSG_HANDSHAKE);
+                sendRendezvousRejection(serv_addr, (reqpkt));
+
+                break; // unure if this should stay. This prevents from sending additionally SHUTDOWN message
```

Comment:
Rendezvous connection, blocking mode.
Current master:
- HS Rejection - not sent,
- SHUTDOWN - not sent.

This PR:
- HS Rejection - sent, 
- SHUTDOWN - not sent.

Reply:
Yes, and the question is whether it is still better that both packets are sent to maintain consistent behavior.

Reply:
I assume SHUTDOWN should only be sent by the initiator on CONCLUSION Response.

Reply:
Well, that makes sense, but on the other hand, the only party that is going do any "checks and possibly rejection" is the responder party. If the initiator party had any reason to reject the already accepted handshake of the peer, this shutdown is likely needed, as the handshake packet will be discarded (only agreement is expected, if so).